pkg/workspace/controllers/workspace_gc_finalizer.go

```diff
@@ -48,3 +59,101 @@ func (c *WorkspaceReconciler) garbageCollectWorkspace(ctx context.Context, wObj
 
 	return ctrl.Result{}, nil
 }
+
+// garbagePersistentVolume ensures that all prersistent volumes are deleted
+func (c *WorkspaceReconciler) garbagePersistentVolume(ctx context.Context, wObj *kaitov1beta1.Workspace) (ctrl.Result, error) {
+	klog.InfoS("garbagePersistentVolume", "workspace", klog.KObj(wObj))
+
+	// List PVCs in the same namespace with a label matching workspace name
+	pvcList := &corev1.PersistentVolumeClaimList{}
+	listOpts := []client.ListOption{
+		client.InNamespace(wObj.Namespace),
+		client.MatchingLabels{kaitov1beta1.LabelWorkspaceName: wObj.Name},
+	}
+
+	if err := c.List(ctx, pvcList, listOpts...); err != nil {
+		klog.ErrorS(err, "failed to list PVCs for workspace", "workspace", klog.KObj(wObj))
+		return ctrl.Result{}, err
+	}
+
+	if len(pvcList.Items) == 0 {
+		klog.InfoS("No PVCs found for workspace", "workspace", klog.KObj(wObj))
+		return ctrl.Result{}, nil
+	}
+
+	// Process the found PVCs
+	var errs error
+	localpvCount := 0
+	for i := range pvcList.Items {
+		pvc := &pvcList.Items[i]
+		if pvc.Spec.StorageClassName != nil && *pvc.Spec.StorageClassName == consts.LocalNVMeStorageClass {
+			localpvCount++
+			if err := deleteOrphanedLocalPV(ctx, c.Client, pvc); err != nil {
+				errs = multierr.Append(errs, err)
+			}
+		}
+	}
+
+	if errs != nil {
+		return ctrl.Result{Requeue: true}, errs
+	}
+
+	if localpvCount > 0 {
+		return ctrl.Result{Requeue: true, RequeueAfter: 5 * time.Second}, nil
+	}
+	return ctrl.Result{}, nil
+}
+
+// When a node with local volumes gets removed from a cluster before deleting those volumes,
+// the PV and PVC objects may still exist. Force deleting objects in this case.
+// Ref: https://github.com/kubernetes-csi/external-provisioner?tab=readme-ov-file#deleting-local-volumes-after-a-node-failure-or-removal
+func deleteOrphanedLocalPV(ctx context.Context, c client.Client, pvc *corev1.PersistentVolumeClaim) error {
+	nodeName := pvc.Annotations["volume.kubernetes.io/selected-node"]
```

Comment:
nit: import the constant from https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/component-helpers/storage/volume/pv_helpers.go#L51

Reply:
done